tests/conftest.py

```diff
@@ -392,6 +394,74 @@ def random_date_generator(start_date, range_in_days):
     return train, test
 
 
+@pytest.fixture
+def dataset_ncf_files(dataset_ncf):
+    train, test = dataset_ncf
+    test = test[test["userID"].isin(train["userID"].unique())]
+    test = test[test["itemID"].isin(train["itemID"].unique())]
+    train = train.sort_values(by=DEFAULT_USER_COL)
+    test = test.sort_values(by=DEFAULT_USER_COL)
+    leave_one_out_test = test.groupby("userID").last().reset_index()
+    return train, test, leave_one_out_test
+
+
+@pytest.fixture
+def data_paths(tmp_path):
+    train_path = os.path.join(tmp_path, "train.csv")
+    test_path = os.path.join(tmp_path, "test.csv")
+    leave_one_out_test_path = os.path.join(tmp_path, "leave_one_out_test.csv")
+    return train_path, test_path, leave_one_out_test_path
+
+
+@pytest.fixture
+def dataset_ncf_files_sorted(data_paths, dataset_ncf_files):
+    train_path, test_path, leave_one_out_test_path = data_paths
+    train, test, leave_one_out_test = dataset_ncf_files
+    train.to_csv(train_path, index=False)
+    test.to_csv(test_path, index=False)
+    leave_one_out_test.to_csv(leave_one_out_test_path, index=False)
+    return train_path, test_path, leave_one_out_test_path
+
+
+@pytest.fixture
+def dataset_ncf_files_unsorted(data_paths, dataset_ncf_files):
+    train_path, test_path, leave_one_out_test_path = data_paths
+    train, test, leave_one_out_test = dataset_ncf_files
+    train = train.sort_values(by=DEFAULT_ITEM_COL)
```

Comment:
Because if we leave the data frame as is there is an infinitesimally small chance the data is sampled in user order. Sorting by item guarantees it's not sorted by user. But I can remove this if you think it just adds confusion?

Reply:
Ah, it's a bit unclear. Maybe make it more explicit e.g. move the highest user to the first position.